### PR TITLE
Delete samples and update README.md

### PR DESCRIPTION
**Pull request by IEvangelist**
- In upgrading these, I realized that I had to manually sync changes to the `dotnet/samples` repo, which begs the question, why not just have the one source of truth?
- Delete samples.
- Update _README.md_ to point to both:

  - :octocat: [dotnet/samples](https://github.com/dotnet/samples/tree/main/orleans)
  - :eyes: [Samples browser](https://learn.microsoft.com/samples/browse/?expanded=dotnet&products=dotnet-orleans)

Related to https://github.com/dotnet/samples/issues/5496

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/orleans/pull/8154)

### CODE REVIEW

File: samples/README.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Orleans Samples
 
+📢 This collection of samples has been moved to the official `dotnet/samples` repository and is part of the Samples browser experience.
+
+- :octocat: [dotnet/samples](https://github.com/dotnet/samples/tree/main/orleans)
+- :eyes: [Samples browser](https://learn.microsoft.com/samples/browse/?expanded=dotnet&products=dotnet-orleans)
+
 ## [Hello, World!](./HelloWorld/#readme)
```
Comment (ReubenBond) [inline samples/README.md:8]: These relative links will be broken now

Comment (IEvangelist) [reply samples/README.md:8]: Good catch, I'll update those.

**Comment by IEvangelist** [comment]
Wait to merge this until #8035 is merged. Rationale in https://github.com/dotnet/orleans/issues/8035#issuecomment-1317468317

**Review by CamSoper — APPROVED** [review]
LGTM

**Pull request merged**